### PR TITLE
Update crossterm version to 0.26

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

~~**WARN**: do not merge this yet, it's using my fork of reedline and lscolors~~

This pr is a companion to https://github.com/nushell/reedline/pull/560

Fortunally, we don't need to change too much nushell code.

## Additional note about lscolor dependency
~~Currently lscolor is using 0.25 version, I've submitted a pr to update it: https://github.com/sharkdp/lscolors/pull/58~~
lscolor is using 0.26 for now

# User-Facing Changes

None

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

> **Note**
> from `nushell` you can also use the `toolkit` as follows
> ```bash
> use toolkit.nu  # or use an `env_change` hook to activate it automatically
> toolkit check pr
> ```

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by fdncred** [comment]
@WindSoilder are we ready to land this? I'd like to get some good testing in before the release if possible.

**Comment by WindSoilder** [comment]
Yeah I think so, I have done some manually testing and think it's ok

File: Cargo.toml [hunk 1]
```diff
@@ -65,7 +65,7 @@ nu-std = { path = "./crates/nu-std", version = "0.78.1" }
 nu-utils = { path = "./crates/nu-utils", version = "0.78.1" }
 
 nu-ansi-term = "0.47.0"
-reedline = { version = "0.18.0", features = ["bashisms", "sqlite"] }
```
Comment (sholderbach) [inline Cargo.toml:68]: Can you revert the manual override outside the patch section? (see L163)

File: crates/nu-cli/Cargo.toml [hunk 2]
```diff
@@ -24,11 +24,11 @@ nu-utils = { path = "../nu-utils", version = "0.78.1" }
 nu-color-config = { path = "../nu-color-config", version = "0.78.1" }
 
 nu-ansi-term = "0.47.0"
-reedline = { version = "0.18.0", features = ["bashisms", "sqlite"] }
```
Comment (sholderbach) [inline crates/nu-cli/Cargo.toml:27]: Same

Comment (WindSoilder) [reply crates/nu-cli/Cargo.toml:27]: Hi, I get a warning message:
warning: patch for the non root package will be ignored, specify patch at the workspace root

It seems that we don't need to change `nu-cli/Cargo.toml`.

File: Cargo.toml (hunk 1 above)
Comment (WindSoilder) [reply Cargo.toml:68]: It's done, would you please have a look again?  Not really sure if I'm doing right thing

**Comment by WindSoilder** [comment]
oh.....ci freeze on pulling tool chain...

**Comment by fdncred** [comment]
I'm scared to push the button. 😆

**Comment by rgwood** [comment]
Thanks for all your work on this @WindSoilder!

**Pull request merged**